File: src/main/java/de/swt/testing/OrderProcessor.java
package de.swt.testing;

/**
 * Processes orders and sends confirmation emails.
 */
public class OrderProcessor {

    private final EmailService emailService;

    public OrderProcessor(EmailService emailService) {
        this.emailService = emailService;
    }

    /**
     * Processes an order and sends a confirmation email.
     *
     * @param customerEmail the customer's email address
     * @param amount the order amount
     * @return true if the order was processed and the email was sent
     */
    public boolean processOrder(String customerEmail, double amount) {
        String subject = "Order Confirmation";
        String body = "Your order of %.2f EUR has been processed.".formatted(amount);
        return emailService.sendEmail(customerEmail, subject, body);
    }
}

File: python/src/order_processor.py
"""Processes orders and sends confirmation emails."""


class OrderProcessor:
    """Processes orders and sends confirmation notifications.

    The notification_service must be any object with a
    ``notify(to, subject, body)`` method that returns a bool.
    """

    def __init__(self, notification_service):
        self._notification_service = notification_service

    def process_order(self, customer_email, amount):
        """Process an order and send a confirmation notification.

        Args:
            customer_email: The customer's email address.
            amount: The order amount.

        Returns:
            True if the order was processed and the notification was sent.
        """
        subject = "Order Confirmation"
        body = f"Your order of {amount:.2f} EUR has been processed."
        return self._notification_service.notify(customer_email, subject, body)
